client/lib/login_register/cubit.dart

```diff
@@ -0,0 +1,149 @@
+import 'package:ccquarters/model/user.dart';
+import 'package:ccquarters/services/auth/service.dart';
+import 'package:ccquarters/services/auth/sign_in_result.dart';
+import 'package:ccquarters/services/auth/sign_up_result.dart';
+import 'package:ccquarters/services/users/service.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:internet_connection_checker/internet_connection_checker.dart';
+
+import 'state.dart';
+
+class AuthCubit extends Cubit<AuthState> {
+  AuthCubit({required this.authService, required this.userService})
+      : super(kIsWeb ? SignedInState() : NeedsSigningInState()) {
+    if (kIsWeb) {
+      user = null;
+      authService.signInAnnonymously();
+      setTokens();
+    } else {
+      user = User();
+    }
+  }
+
+  AuthService authService;
+  UserService userService;
+  User? user;
```

Comment:
Do we need nullability here? Maybe late is enough?

Reply:
I made nullability here to have null when user is not logged in, I can also make property in User class